### ISSUE TITLE
1.1.334 regression: `files` cli argument not respected?

### ISSUE THREAD

**Issue by Avasam** [microsoft/pyright#6311]
**Describe the bug**
Running pyright 1.1.334 locally on typeshed for a specific folder gives me errors for files I haven't specified.

**Code or Screenshots**
```shell
> npx pyright@1.1.333 stubs/openpyxl
0 errors, 0 warnings, 0 informations
```

```shell
> npx pyright@1.1.334 stubs/openpyxl
E:\Users\Avasam\Documents\Git\typeshed\stubs\Flask-Migrate\flask_migrate\__init__.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\Flask-Migrate\flask_migrate\__init__.pyi:7:6 - error: Import "flask_sqlalchemy" could not be resolved (reportMissingImports)
E:\Users\Avasam\Documents\Git\typeshed\stubs\aiofiles\aiofiles\os.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\aiofiles\aiofiles\os.pyi:9:29 - error: "wrap" is unknown import symbol (reportGeneralTypeIssues)
E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:48:63 - error: Expected no type arguments for class "RedisClusterCommands" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:48:82 - error: Type argument for "Generic" must be a type variable
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:8:62 - error: "_ParseResponseOptions" is unknown import symbol (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:9:28 - error: "CommandsParser" is unknown import symbol (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:10:33 - error: "_StrType" is unknown import symbol (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:11:30 - error: "BaseParser" is unknown import symbol (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:11:70 - error: "Encoder" is not exported from module "redis.connection"
    Import from "encoders" instead (reportPrivateImportUsage)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:11:79 - error: "_ConnectionPoolOptions" is unknown import symbol (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:11:103 - error: "_Encodable" is unknown import symbol (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:17:38 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:33:21 - error: Base class type is unknown, obscuring type of derived class (reportUntypedBaseClass) 
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:58:53 - error: Expected no type arguments for class "CaseInsensitiveDict" (reportGeneralTypeIssues) 
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:59:43 - error: Expected no type arguments for class "CaseInsensitiveDict" (reportGeneralTypeIssues) 
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:85:64 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:123:29 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:125:93 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:167:85 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:175:27 - error: Expected no type arguments for class "RedisCluster" (reportGeneralTypeIssues)       
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:178:37 - error: Expected no type arguments for class "RedisCluster" (reportGeneralTypeIssues)       
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:185:37 - error: Expected no type arguments for class "RedisCluster" (reportGeneralTypeIssues)       
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:189:45 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\cluster.pyi:191:36 - error: Expected no type arguments for class "RedisCluster" (reportGeneralTypeIssues)       
E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\lock.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\lock.pyi:18:18 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\lock.pyi:28:22 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\sentinel.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\sentinel.pyi:10:42 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\sentinel.pyi:40:27 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\sentinel.pyi:55:94 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\sentinel.pyi:59:93 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:12:41 - error: "_CommandOptions" is unknown import symbol (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:12:58 - error: "_Key" is unknown import symbol (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:12:64 - error: "_StrType" is unknown import symbol (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:12:74 - error: "_Value" is unknown import symbol (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:27:67 - error: Expected no type arguments for class "AsyncCoreCommands" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:27:109 - error: Type argument for "Generic" must be a type variable
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:70:16 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:108:16 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:111:21 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:147:21 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:292:22 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:185:93 - error: Expected no type arguments for class "Pipeline" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:188:34 - error: Expected no type arguments for class "Pipeline" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:206:31 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:319:60 - error: Expected no type arguments for class "Pipeline" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:319:91 - error: Expected no type arguments for class "Pipeline" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\client.pyi:358:19 - error: Expected no type arguments for class "Pipeline" (reportGeneralTypeIssues)
E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\cluster.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\cluster.pyi:8:38 - error: "BaseParser" is not exported from module "redis.asyncio.connection"
    Import from "base" instead (reportPrivateImportUsage)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\cluster.pyi:8:62 - error: "Encoder" is not exported from module "redis.asyncio.connection"
    Import from "encoders" instead (reportPrivateImportUsage)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\cluster.pyi:15:33 - error: "_StrType" is unknown import symbol (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\cluster.pyi:23:65 - error: Type argument for "Generic" must be a type variable
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\cluster.pyi:155:68 - error: Type argument for "Generic" must be a type variable
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\cluster.pyi:100:106 - error: Expected no type arguments for class "ClusterPipeline" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\cluster.pyi:156:45 - error: Expected no type arguments for class "RedisCluster" (reportGeneralTypeIssues)
E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\lock.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\lock.pyi:17:18 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\lock.pyi:27:22 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\sentinel.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\sentinel.pyi:59:33 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\sentinel.pyi:66:33 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\utils.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\utils.pyi:5:26 - error: "_StrType" is unknown import symbol (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\utils.pyi:7:43 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\utils.pyi:9:24 - error: Type argument for "Generic" must be a type variable
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\utils.pyi:10:17 - error: Expected no type arguments for class "Pipeline" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\utils.pyi:11:41 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\asyncio\utils.pyi:12:44 - error: Expected no type arguments for class "Pipeline" (reportGeneralTypeIssues)
E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\commands\core.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\commands\core.pyi:1458:54 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\redis\redis\commands\core.pyi:1460:112 - error: Expected no type arguments for class "Redis" (reportGeneralTypeIssues)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\axisgrid.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\axisgrid.pyi:6:6 - error: Stub file not found for "matplotlib.artist" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\axisgrid.pyi:7:6 - error: Stub file not found for "matplotlib.axes" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\axisgrid.pyi:8:6 - error: Stub file not found for "matplotlib.colors" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\axisgrid.pyi:9:6 - error: Stub file not found for "matplotlib.figure" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\axisgrid.pyi:10:6 - error: Stub file not found for "matplotlib.legend" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\axisgrid.pyi:11:6 - error: Stub file not found for "matplotlib.text" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\axisgrid.pyi:12:6 - error: Import "matplotlib.typing" could not be resolved (reportMissingImports)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\categorical.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\categorical.pyi:6:6 - error: Stub file not found for "matplotlib.axes" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\categorical.pyi:7:6 - error: Import "matplotlib.typing" could not be resolved (reportMissingImports)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\cm.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\cm.pyi:1:6 - error: Stub file not found for "matplotlib.colors" (reportMissingTypeStubs)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\distributions.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\distributions.pyi:6:6 - error: Stub file not found for "matplotlib.axes" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\distributions.pyi:7:6 - error: Stub file not found for "matplotlib.colors" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\distributions.pyi:8:6 - error: Import "matplotlib.typing" could not be resolved (reportMissingImports)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\matrix.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\matrix.pyi:6:6 - error: Stub file not found for "matplotlib.axes" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\matrix.pyi:7:6 - error: Stub file not found for "matplotlib.colors" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\matrix.pyi:8:6 - error: Stub file not found for "matplotlib.gridspec" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\matrix.pyi:9:6 - error: Import "matplotlib.typing" could not be resolved (reportMissingImports)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\miscplot.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\miscplot.pyi:3:6 - error: Import "matplotlib.typing" could not be resolved (reportMissingImports)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\palettes.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\palettes.pyi:5:6 - error: Stub file not found for "matplotlib.colors" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\palettes.pyi:6:6 - error: Import "matplotlib.typing" could not be resolved (reportMissingImports)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\rcmod.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\rcmod.pyi:5:6 - error: Import "matplotlib.typing" could not be resolved (reportMissingImports)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\regression.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\regression.pyi:7:6 - error: Stub file not found for "matplotlib.axes" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\regression.pyi:8:6 - error: Import "matplotlib.typing" could not be resolved (reportMissingImports)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\relational.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\relational.pyi:6:6 - error: Stub file not found for "matplotlib.axes" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\relational.pyi:7:6 - error: Stub file not found for "matplotlib.colors" (reportMissingTypeStubs)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\utils.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\utils.pyi:7:6 - error: Stub file not found for "matplotlib.axes" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\utils.pyi:8:6 - error: Stub file not found for "matplotlib.figure" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\utils.pyi:9:6 - error: Stub file not found for "matplotlib.legend" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\utils.pyi:10:6 - error: Stub file not found for "matplotlib.text" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\utils.pyi:11:6 - error: Stub file not found for "matplotlib.ticker" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\utils.pyi:12:6 - error: Import "matplotlib.typing" could not be resolved (reportMissingImports)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\widgets.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\widgets.pyi:4:6 - error: Stub file not found for "matplotlib.colors" (reportMissingTypeStubs)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\plot.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\plot.pyi:9:8 - error: Stub file not found for "matplotlib" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\plot.pyi:10:6 - error: Stub file not found for "matplotlib.axes" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\plot.pyi:11:6 - error: Stub file not found for "matplotlib.figure" (reportMissingTypeStubs)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\properties.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\properties.pyi:6:6 - error: Stub file not found for "matplotlib.markers" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\properties.pyi:7:6 - error: Stub file not found for "matplotlib.path" (reportMissingTypeStubs)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\scales.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\scales.pyi:7:6 - error: Stub file not found for "matplotlib.axis" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\scales.pyi:8:6 - error: Stub file not found for "matplotlib.scale" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\scales.pyi:9:6 - error: Stub file not found for "matplotlib.ticker" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\scales.pyi:10:6 - error: Stub file not found for "matplotlib.units" (reportMissingTypeStubs)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\subplots.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\subplots.pyi:4:6 - error: Stub file not found for "matplotlib.axes" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\subplots.pyi:5:6 - error: Stub file not found for "matplotlib.figure" (reportMissingTypeStubs)
E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\typing.pyi
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\typing.pyi:7:6 - error: Stub file not found for "matplotlib.colors" (reportMissingTypeStubs)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\typing.pyi:12:41 - error: Expected no type arguments for class "Index" (reportGeneralTypeIssues)
  E:\Users\Avasam\Documents\Git\typeshed\stubs\seaborn\seaborn\_core\typing.pyi:14:58 - error: Expected no type arguments for class "Index" (reportGeneralTypeIssues)
112 errors, 0 warnings, 0 informations
```

**VS Code extension or command-line**
`npx pyright@1.1.334`

**Comment by erictraut** [comment]
Thanks for the bug report. Yes, this is a regression caused by a change in how we compose configuration options from the command-line interface, language server settings, and config file options. In most cases, the config file options (i.e. those found in the `pyrightconfig.json` or `pyproject.toml` files) override settings provided on the command line or language server settings. In this example, there is a `pyrightconfig.json` file present, and it defines an "includes" config setting. This is taking precedent over the files you're passing on the command line. Prior to pyright 1.1.334, the files passed on the command line too precedence.

I think it makes sense to go back to the previous behavior, at least partly. It's not clear whether the "exclude" setting in the config file should be applied on top of the files specified on the command line. Probably not. Previously, it did.

**Comment by Avasam** [comment]
>  It's not clear whether the "exclude" setting in the config file should be applied on top of the files specified on the command line.

Current typeshed tooling currently assumes it does. The two use-cases are:
- type-checking stubs, and running regression tests from the `test_cases` files, are two distinct test suites. So `pyright stubs/setuptools` for example shouldn't check test_cases.
- Running `pyright stdlib --project pyrightconfig.stricter.json` shouldn't run on files excluded from the stricter config (there used to be stubs with partial stricter configs as well, but atm it's only stdlib). Otherwise it would lead to unexpected errors in CLI.

So if this changes, typeshed should update how their tests are run if possible to align with the new config expectations.

**Comment by erictraut** [comment]
OK, I'm going to restore the old behavior then.

It's a bit odd for pyright to honor the "exclude" setting in the config file but override the "include" setting with the `--files` CLI option. This behavior wasn't intended. But if typeshed is relying on this, I'll leave it as is for now.

**Comment by Avasam** [comment]
FWIW, typeshed *can* update its pyright scripts and configs if it was relying on unintended behaviour (by reading the specified config file for the exclude setting and appending it to the pyright command in `tests/pyright_test.py`) if you believe that's the way to go forward.

Speaking of which, I'll CC @sobolevn who usually bumps the pyright version in typeshed.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.335, which I just published.
